test/mitmproxy/addons/test_dns_resolver.py

```diff
@@ -32,107 +25,86 @@ async def test_simple(monkeypatch):
-                        socket.AF_INET6,
-                        type,
-                        None,
-                        None,
-                        ("2001:4860:4860::8888", port, None, None),
-                    )
-                ]
-        else:
-            e.errno = socket.EAI_FAMILY
-        raise e
-
-
-async def test_resolve():
-    async def fail_with(question: dns.Question, code: int):
-        with pytest.raises(dns_resolver.ResolveError) as ex:
-            await dns_resolver.resolve_question(question, DummyLoop())
-        assert ex.value.response_code == code
-
-    async def succeed_with(
-        question: dns.Question, check: Callable[[dns.ResourceRecord], bool]
-    ):
-        assert any(
-            map(check, await dns_resolver.resolve_question(question, DummyLoop()))
-        )
+async def test_resolver():
+    dr = dns_resolver.DnsResolver()
+    with taddons.context(dr) as tctx:
+        assert dr.name_servers == mitmproxy_rs.get_system_dns_servers()
 
-    await fail_with(
-        dns.Question("dns.google", dns.types.A, dns.classes.CH),
-        dns.response_codes.NOTIMP,
-    )
-    await fail_with(
-        dns.Question("not.exists", dns.types.A, dns.classes.IN),
-        dns.response_codes.NXDOMAIN,
-    )
-    await fail_with(
-        dns.Question("dns.google", dns.types.SOA, dns.classes.IN),
-        dns.response_codes.NOTIMP,
-    )
-    await fail_with(
-        dns.Question("totally.invalid", dns.types.PTR, dns.classes.IN),
-        dns.response_codes.FORMERR,
-    )
-    await fail_with(
-        dns.Question("invalid.in-addr.arpa", dns.types.PTR, dns.classes.IN),
-        dns.response_codes.FORMERR,
-    )
-    await fail_with(
-        dns.Question("0.0.0.1.in-addr.arpa", dns.types.PTR, dns.classes.IN),
-        dns.response_codes.NXDOMAIN,
-    )
+        tctx.options.name_servers = ["1.1.1.1"]
+        assert dr.name_servers == ["1.1.1.1"]
 
-    await succeed_with(
-        dns.Question("dns.google", dns.types.A, dns.classes.IN),
-        lambda rr: rr.ipv4_address == ipaddress.IPv4Address("8.8.8.8"),
-    )
-    await succeed_with(
-        dns.Question("dns.google", dns.types.AAAA, dns.classes.IN),
-        lambda rr: rr.ipv6_address == ipaddress.IPv6Address("2001:4860:4860::8888"),
-    )
-    await succeed_with(
-        dns.Question("8.8.8.8.in-addr.arpa", dns.types.PTR, dns.classes.IN),
-        lambda rr: rr.domain_name == "dns.google",
-    )
-    await succeed_with(
-        dns.Question(
-            "8.8.8.8.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.6.8.4.0.6.8.4.1.0.0.2.ip6.arpa",
-            dns.types.PTR,
-            dns.classes.IN,
-        ),
-        lambda rr: rr.domain_name == "dns.google",
+        res_old = dr.resolver
+        tctx.options.use_hosts_file = False
+        assert dr.resolver != res_old
+
+
+async def lookup_ipv4(name: str):
+    if name == "not.exists":
+        raise socket.gaierror("NXDOMAIN")
+    return await asyncio.sleep(0, ["8.8.8.8"])
```

Comment:
Not sure if I want to know - what is this sleep here about? 😄

Reply:
oops, my bad